strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/services/ArtifactEntryService.java

```diff
@@ -0,0 +1,30 @@
+package org.carlspring.strongbox.services;
+
+import org.carlspring.strongbox.artifact.coordinates.ArtifactCoordinates;
+import org.carlspring.strongbox.data.service.CrudService;
+import org.carlspring.strongbox.domain.ArtifactEntry;
+
+import java.util.List;
+
+import org.springframework.transaction.annotation.Transactional;
+
+/**
+ * CRUD service for managing {@link ArtifactEntry} entities.
+ *
+ * @author Alex Oreshkevich
+ */
+@Transactional
+public interface ArtifactEntryService
+        extends CrudService<ArtifactEntry, String>
+{
+
+    /**
+     * Returns list of artifacts that matches search query defined as {@link ArtifactCoordinates} fields.
+     * By default all fields are optional and combined using logical AND operator.
+     * If all coordinates aren't present this query will delegate request to {@link #findAll()}.
```

Comment:
What does that mean? Could you give an example? If the coordinates have five parameters (`groupId`, `artifactId`, `version`, `type`, `packaging`) and one hasn't been specified, then it will do a `findAll()`?